server/controllers/prisonerCellHistoryController.test.ts

```diff
@@ -0,0 +1,157 @@
+import { PrisonerMockDataA } from '../data/localMockData/prisoner'
+import { inmateDetailMock } from '../data/localMockData/inmateDetailMock'
+import { CaseLoadsDummyDataA } from '../data/localMockData/caseLoad'
+import { prisonApiClientMock } from '../../tests/mocks/prisonApiClientMock'
+import PrisonerCellHistoryController from './prisonerCellHistoryController'
+import { auditServiceMock } from '../../tests/mocks/auditServiceMock'
+import { PrisonApiClient } from '../data/interfaces/prisonApiClient'
+import { AuditService } from '../services/auditService'
+import {
+  OffenderCellHistoryMock,
+  OffenderCellHistoryReceptionMock,
+} from '../data/localMockData/offenderCellHistoryMock'
+import ReceptionsWithCapacityMock from '../data/localMockData/receptionsWithCapacityMock'
+import StaffDetailsMock from '../data/localMockData/staffDetails'
+import AgencyMock from '../data/localMockData/agency'
+import config from '../config'
+
+const profileUrl = `/prisoner/${PrisonerMockDataA.prisonerNumber}`
+const currentLocation: Record<string, unknown> = {
+  agencyId: 'MDI',
+  assignmentDateTime: '2021-07-05T10:35:17',
+  assignmentEndDateTime: '2021-07-05T10:35:17',
+  establishment: null,
+  isTemporaryLocation: false,
+  livingUnitId: 123123,
+  location: '1-1-2',
+  movedIn: '05/07/2021 - 10:35',
+  movedInBy: 'John Smith',
+  movedOut: '05/07/2021 - 10:35',
+}
+const prisonerCellHistoryPageData: Record<string, unknown> = {
+  pageTitle: 'Location details',
+  breadcrumbPrisonerName: 'John Saunders',
+  name: 'John Saunders',
+  prisonerName: 'Saunders, John',
+  prisonerNumber: PrisonerMockDataA.prisonerNumber,
+  profileUrl,
+  canViewCellMoveButton: true,
+  canViewMoveToReceptionButton: true,
+  cellHistoryGroupedByAgency: [],
+  dpsBaseUrl: `${config.apis.dpsHomePageUrl}${profileUrl}`,
+  changeCellLink: `${config.apis.dpsHomePageUrl}${profileUrl}/cell-move/search-for-cell?returnUrl=${profileUrl}`,
+  moveToReceptionLink: `${config.apis.dpsHomePageUrl}${profileUrl}/reception-move/consider-risks-reception`,
+  currentLocation,
+  occupants: [],
+  prisonId: 'MDI',
+}
```

Comment:
The types of these mocks aren't particularly helpful, is there not a real type for them?

Reply:
Agreed, I'm going to tackle this sort of thing in this ticket: https://dsdmoj.atlassian.net/browse/CDPS-373 which is going to be a refactor and implementation of tests for the rest of the page!